src/core/config.c

```diff
@@ -0,0 +1,110 @@
+#include "config.h"
+#include "xdpw.h"
+#include "logger.h"
+
+#include <dictionary.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <iniparser.h>
+
+void print_config(enum LOGLEVEL loglevel, struct xdpw_config *config) {
+	logprint(loglevel, "config: Outputname  %s", config->screencast_conf.output_name);
+}
+
+// NOTE: calling finish_config won't prepare the config to be read again from config file
+// with init_config since to pointers and other values won't be reset to NULL, or 0
+void finish_config(struct xdpw_config *config) {
+	logprint(DEBUG, "config: destroying config");
+
+	// screencast
+	free(&config->screencast_conf.output_name);
+}
+
+static void getstring_from_conffile(dictionary *d,
+		const char *key, char **dest, const char *fallback) {
+	if (*dest != NULL) {
+		return;
+	}
+	const char *c = iniparser_getstring(d, key, fallback);
+	if (c == NULL) {
+		return;
+	}
+	// Allow keys without value as default
+	if (strcmp(c, "") != 0) {
+		*dest = strdup(c);
+	} else {
+		*dest = fallback ? strdup(fallback) : NULL;
+	}
+}
+
+static bool file_exists(const char *path) {
+	return path && access(path, R_OK) != -1;
+}
+
+static char *config_path(char *prefix, char *filename) {
+	if (!prefix || !prefix[0] || !filename || !filename[0]) {
+		return NULL;
+	}
+
+	char *config_folder = "xdg-desktop-portal-wlr";
+
+	size_t size = 3 + strlen(prefix) + strlen(config_folder) + strlen(filename);
+	char *path = calloc(size, sizeof(char));
+	snprintf(path, size, "%s/%s/%s", prefix, config_folder, filename);
+	return path;
+}
+
+static void config_parse_file(const char *configfile, struct xdpw_config *config) {
+	dictionary *d = iniparser_load(configfile);
```

Comment:
We should check for a NULL `d` here, in case `iniparser_load` failed.

Reply:
Same here, if ```d``` is ```NULL``` iniparser_get* will just return the fallback.

Reply:
OK, but if iniparser failed to read the file we want a proper error message

Reply:
Added logging, but don't know how to supress the log from iniparser. We could check before and skip it when the config file is null, or just have the line in the log.

Reply:
Hm, which log from iniparser?

Reply:
There is the line
iniparser: cannot open (NIL) or path to file
in stderr if iniparser_load fails.

Reply:
Maybe don't call `iniparser_load` if `configfile == NULL`?

Reply:
I moved all checks of configfile to the part where we load it with iniparser_load, so we don't have to check for the same things multiple times.